### PR TITLE
langgraph: add support for passing store via state_modifier

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py [hunk 1]
```diff
@@ -54,26 +56,37 @@ class AgentState(TypedDict):
...
             lambda state: [state_modifier] + state["messages"],
             name=STATE_MODIFIER_RUNNABLE_NAME,
         )
     elif callable(state_modifier):
-        state_modifier_runnable = RunnableLambda(
-            state_modifier, name=STATE_MODIFIER_RUNNABLE_NAME
+        # Inspect the state_modifier signature
+        sig = inspect.signature(state_modifier)
+
+        if store is None and "store" in sig.parameters:
```
Comment (nfcampos) [inline libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:82]: remove this, this raises exceptions elsewhere

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py [hunk 2]
```diff
@@ -473,11 +487,16 @@ def should_continue(state: AgentState) -> Literal["tools", "__end__"]:
...
+    # we're passing store here for validation
+    preprocessor = _get_model_preprocessing_runnable(
+        state_modifier, messages_modifier, store
+    )
     model_runnable = preprocessor | model
 
     # Define the function that calls the model
-    def call_model(state: AgentState, config: RunnableConfig) -> AgentState:
+    def call_model(
+        state: AgentState, config: RunnableConfig, *, store: BaseStore
```
Comment (nfcampos) [inline libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:498]: remove

File: libs/langgraph/langgraph/prebuilt/chat_agent_executor.py [hunk 3]
```diff
@@ -495,7 +514,9 @@ def call_model(state: AgentState, config: RunnableConfig) -> AgentState:
         # We return a list, because this will get added to the existing list
         return {"messages": [response]}
 
-    async def acall_model(state: AgentState, config: RunnableConfig) -> AgentState:
+    async def acall_model(
+        state: AgentState, config: RunnableConfig, *, store: BaseStore
```
Comment (nfcampos) [inline libs/langgraph/langgraph/prebuilt/chat_agent_executor.py:518]: remove

**Review by nfcampos — COMMENTED** [review]
remove runnable lambda import

**Review by hinthornw — APPROVED** [review]

**Pull request merged**